frontend/src/cljs/imas_seamap/events.cljs

```diff
@@ -511,11 +511,33 @@
 
 (defn catalogue-toggle-node [{:keys [db]} [_ group nodeid]]
   (let [nodes         (get-in db [:display :catalogue group :expanded])
-        updated-nodes (if (contains? nodes nodeid) (disj nodes nodeid) (conj nodes nodeid))
+        updated-nodes (if (nodes nodeid) (disj nodes nodeid) (conj nodes nodeid))
         db            (assoc-in db [:display :catalogue group :expanded] updated-nodes)]
     {:db       db
      :put-hash (encode-state db)}))
 
+(defn catalogue-add-node [{:keys [db]} [_ group nodeid]]
+  (let [nodes         (get-in db [:display :catalogue group :expanded])
+        updated-nodes (conj nodes nodeid)
+        db            (assoc-in db [:display :catalogue group :expanded] updated-nodes)]
+    {:db       db
+     :put-hash (encode-state db)}))
+
+(defn catalogue-add-nodes-to-layer
+  "Opens nodes in catalogue along path to specified layer"
+  [{:keys [db]} [_ group layer tab categories]]
+  (let [sorting-info (:sorting db)
+        node-ids   (reduce
+                    (fn [node-ids category]
+                      (let [sorting-id (or (get-in sorting-info [category (category layer) 1]) "nil")
```

Comment:
get-in can return a default (not-found) value: https://clojuredocs.org/clojure.core/get-in

Reply:
Can't believe this slipped my mind too.